packages/api-plugin-simple-schema/src/schemas/schema.graphql

```diff
@@ -0,0 +1,19 @@
+"The response from the `introspectSchema` query"
+type IntrospectSchemaPayload {
+  "The schema name"
+  schemaName: String!
+
+  "The JSONObject schema for the schema"
+  schema: JSONObject
+}
+
+extend type Query {
+  "Query the fields of a schema"
+  introspectSchema(
+    "The shop ID of the schema"
```

Comment:
Schemas are not shop specific, so this would be "The shop ID to use for checking permissions". But I think it would be better to use global permissions for this, which means you could remove the `shopId` input.

Reply:
@aldeed @zenweasel 

As per an earlier discussion with Brent, we decided to give access to some common schemas (like Cart, Product, Order, Account, Promotion) for users in "shop manager" or "owner" groups. 
The migration script and the checkAndSeed.js reflects this plan. As per my understanding, this by default makes the permissions/roles to be added under a specific shopId. Meanwhile the global_roles (userPermission.__global_roles__) are picked from the global groups (groups without a shopId, namely "accounts manager" and "system-manager"). 

So if we remove the shopId param in the introspectSchema query, then we have to re-visit the earlier plan. At least we have to include these roles (selected schema access roles) in the global-groups in migration/checkAndSeed code.

Please let me know if I should go ahead with those changes as well.

Reply:
@sujithvn Thanks for the explanation. That does change the situation slightly. But what if there is an accounts manager who also needs to introspect schemas? This person might not belong to any shops. To support every case, I think `shopId` argument should be kept but should be optional. Along with also changing the description to "The shop ID to use for checking permissions" as I mentioned.

Another option would be to make a new default global group "schema viewer", which any user could be added to. Just depends on exactly how all of this will be used over time.

Reply:
@aldeed 
updated as you have suggested. 
- shopId is optional
- provided default permission to `system-manager` also

Did not go for the option of new default group schema-viewer in the interest of time to release this